### PR TITLE
Upgrade spdlog to 1.4 and move to Static lib version

### PR DESCRIPTION
**Pull request by robertpi**
@DataDog/apm-dotnet

### CODE REVIEW

File: src/Datadog.Trace.ClrProfiler.Native/lib/spdlog/include/spdlog/fmt/fmt.h [hunk 1]
```diff
@@ -10,16 +10,18 @@
 // By default spdlog include its own copy.
 //
 
-#if 0 // !defined(SPDLOG_FMT_EXTERNAL)
```
Comment (andrewlock) [inline src/Datadog.Trace.ClrProfiler.Native/lib/spdlog/include/spdlog/fmt/fmt.h:13]: Was this change intentional? Don't have the context for why it was set like this previously

File: src/Datadog.Trace.ClrProfiler.Native/lib/spdlog/include/spdlog/fmt/ostr.h [hunk 2]
```diff
@@ -7,7 +7,7 @@
 //
 // include bundled or external copy of fmtlib's ostream support
 //
-#if 0 // !defined(SPDLOG_FMT_EXTERNAL)
+#if !defined(SPDLOG_FMT_EXTERNAL)
```
Comment (andrewlock) [inline src/Datadog.Trace.ClrProfiler.Native/lib/spdlog/include/spdlog/fmt/ostr.h:10]: Same for this one

**Review by andrewlock — APPROVED** [review]
Not sure about the implications for this, so I'll defer to others but LGTM

**Review by tonyredondo — APPROVED** [review]
LGTM
